profiler/src/main/java/com/splunk/opentelemetry/profiler/JfrSettingsOverrides.java

```diff
@@ -40,25 +42,21 @@
 
   Map<String, String> apply(Map<String, String> jfrSettings) {
     Map<String, String> settings = new HashMap<>(jfrSettings);
-    jfrSettings.keySet().stream()
-        .filter(key -> key.endsWith("#period"))
-        .forEach(
-            key -> {
-              String[] parts = key.split("#");
-              String eventName = parts[0];
-              String shortEventName = eventName.replaceFirst("^jdk.", "");
-              String configKey = CONFIG_KEY_PERIOD_PREFIX + "." + shortEventName.toLowerCase();
-              String customSetting = config.getString(configKey);
-              if (customSetting != null) {
-                String jfrFormattedDuration = customSetting + " ms";
-                logger.info(
-                    "Custom JFR period configured for {} :: {}", eventName, jfrFormattedDuration);
-                settings.put(key, jfrFormattedDuration);
-              }
-            });
+    Duration customInterval = getCustomInterval();
+    if (customInterval != Duration.ZERO) {
+      settings.put("jdk.ThreadDump#period", customInterval.toMillis() + " ms");
+    }
     return maybeEnableTLABs(settings);
   }
 
+  private Duration getCustomInterval() {
+    Duration customInterval = config.getDuration(CONFIG_KEY_CALL_STACK_INTERVAL, Duration.ZERO);
+    if (customInterval != Duration.ZERO) {
+      return customInterval;
+    }
+    return config.getDuration(CONFIG_KEY_DEPRECATED_THREADDUMP_PERIOD, Duration.ZERO);
```

Comment:
Should we log some warning if deprecated config is still used?

Reply:
Sure, good idea.